src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -355,20 +366,23 @@ export class HttpConnection implements IConnection {
                 transportExceptions.push(`${endpoint.transport} failed: ${transportOrError}`);
             } else if (this.isITransport(transportOrError)) {
                 this.transport = transportOrError;
-                if (!negotiateResponse.connectionId) {
+                if (!negotiateResponse.connectionToken) {
                     try {
                         negotiateResponse = await this.getNegotiationResponse(url);
+                        this.connectionId = negotiateResponse.connectionId;
                     } catch (ex) {
                         return Promise.reject(ex);
                     }
-                    connectUrl = this.createConnectUrl(url, negotiateResponse.connectionId);
+                    connectUrl = this.createConnectUrl(url, negotiateResponse.connectionToken);
                 }
                 try {
                     await this.startTransport(connectUrl, requestedTransferFormat);
                     return;
                 } catch (ex) {
                     this.logger.log(LogLevel.Error, `Failed to start the transport '${endpoint.transport}': ${ex}`);
+                    negotiateResponse.connectionToken = undefined;
```

Comment:
nit: It might be a little cleaner to unset negotiateResponse entirely, and check `if (negotiateResponse !== undefined)` instead of `if (!negotiateResponse.connectionToken)`.